gateway/src/main/java/io/zeebe/gateway/impl/job/LongPollingActivateJobsHandler.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.0. You may not use this file
+ * except in compliance with the Zeebe Community License 1.0.
+ */
+package io.zeebe.gateway.impl.job;
+
+import io.atomix.cluster.messaging.Subscription;
+import io.grpc.stub.StreamObserver;
+import io.zeebe.gateway.Loggers;
+import io.zeebe.gateway.impl.broker.BrokerClient;
+import io.zeebe.gateway.impl.broker.cluster.BrokerClusterState;
+import io.zeebe.gateway.protocol.GatewayOuterClass.ActivateJobsRequest;
+import io.zeebe.gateway.protocol.GatewayOuterClass.ActivateJobsResponse;
+import io.zeebe.util.sched.Actor;
+import io.zeebe.util.sched.ScheduledTimer;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.CompletableFuture;
+import org.slf4j.Logger;
+
+public class LongPollingActivateJobsHandler extends Actor {
+
+  public static final Duration DEFAULT_LONG_POLLING_TIMEOUT = Duration.ofSeconds(10);
+  private static final String JOBS_AVAILABLE_TOPIC = "jobsAvailable";
+  private static final int NO_JOBS_RESPONSE_THRESHOLD = 3;
```

Comment:
I think it would make sense that users can configure the threshold. What do you think?